packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx

```diff
@@ -176,25 +179,62 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
 
   const focusDateRef = React.useRef(null);
 
+  const contstraintNavigatedDate = (initialDate: Date, targetDate: Date, direction: number) => {
+    if (!targetDate) {
+      // if we couldn't find a target date at all, do nothing
+      return undefined;
+    }
+
+    const findAvailableDateOptions: IAvailableDateOptions = {
+      initialDate,
+      targetDate,
+      direction,
+      restrictedDates,
+      minDate,
+      maxDate,
+    };
+
+    let newNavigatedDate = findAvailableDate(findAvailableDateOptions);
+
+    if (!newNavigatedDate) {
+      // if no dates available in initial direction, try going backwards
+      findAvailableDateOptions.direction = -direction;
+      newNavigatedDate = findAvailableDate(findAvailableDateOptions);
+    }
+
+    return newNavigatedDate;
+  };
+
   const changeMonth = (nextMonth: boolean) => {
+    const direction = nextMonth ? 1 : -1;
     const updatedGridNavigatedDate = addMonths(gridNavigatedDate, nextMonth ? 1 : -1);
-    setGridNavigatedDate(updatedGridNavigatedDate);
+
+    const newNavigatedDate = contstraintNavigatedDate(gridNavigatedDate, updatedGridNavigatedDate, direction);
+
+    if (newNavigatedDate) {
+      setGridNavigatedDate(newNavigatedDate);
+    }
   };
 
+  const prevMonthOutOfBounds = minDate ? compareDatePart(minDate, getMonthStart(gridNavigatedDate)) >= 0 : false;
+  const nextMonthOutOfBounds = maxDate ? compareDatePart(getMonthEnd(gridNavigatedDate), maxDate) >= 0 : false;
+
   const handleKeyDown = (e, day) => {
     const keyCode = getCode(e);
     const initialDate = day.originalDate;
     let targetDate: Date | null = null;
+    const visuallyPleasingWeekJumpNavigation = false;
```

Comment:
```suggestion
    const skipWeekOnArrowNavigation = false;
```
?

Reply:
Yeah, this proposal is more descriptive 👍 :) And less cryptic.